crates/hir/src/semantics.rs

```diff
@@ -1145,9 +1145,9 @@ to_def_impls![
     (crate::Field, ast::RecordField, record_field_to_def),
     (crate::Field, ast::TupleField, tuple_field_to_def),
     (crate::Variant, ast::Variant, enum_variant_to_def),
-    (crate::TypeParam, ast::TypeParam, type_param_to_def),
+    (crate::TypeOrConstParam, ast::TypeParam, type_param_to_def),
     (crate::LifetimeParam, ast::LifetimeParam, lifetime_param_to_def),
-    (crate::ConstParam, ast::ConstParam, const_param_to_def),
+    (crate::TypeOrConstParam, ast::ConstParam, const_param_to_def),
```

Comment:
I wonder whether it makes sense to change the ToDef targets here, since we still have the distinct hir types

Reply:
Is it now good?